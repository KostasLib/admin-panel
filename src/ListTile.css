.list-tile {
  height: 6rem;
  display: flex;
  position: relative;
  cursor: pointer;
  z-index: 1;
}

/* Highlight on hover */
.list-tile::before {
  /* Positioning */
  content: "";
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;

  /* Styles */
  background-color: #e8e8e8;
  opacity: 0;
  z-index: -1;
  visibility: hidden;
  transition: all 0.1s linear;
}

.list-tile:hover::before {
  opacity: 1;
  visibility: visible;
}

.list-tile__avatar {
  width: 7rem;
  height: 7rem;
  background-color: lightpink;
}

.list-tile__details {
  flex: 1;
  display: none;
  background-color: lightgreen;
}

/* Show element on desktop */
@media screen and (min-width: 768px) {
  .list-tile__details {
    display: block;
  }
}
