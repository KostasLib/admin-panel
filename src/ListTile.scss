@import 'variables';

.list-tile {
  height: $list-tile-height;
  display: flex;
  position: relative;
  cursor: pointer;
  z-index: 1;

  /* Pseudo-element for highlighting */
  &::before {
    // Position
    content: '';
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: -1;

    // Styles
    background-color: $user-hovered-bg;
    opacity: 0;
    visibility: hidden;
    transition: all 0.1s linear;
  }

  // Modifier to show the highlight pseudo-el.
  &--with-hover {
    &:hover::before {
      opacity: 1;
      visibility: visible;
    }
  }

  // Modifier to change pseudo's color and show it steadily when selected.
  &--selected {
    &::before {
      background-color: $user-selected-bg;
      opacity: 1;
      visibility: visible;
    }
  }

  // Avatar el
  &__avatar {
    width: $avatar-width;
  }

  // Details el
  &__details {
    flex: 1;
    display: none;

    // Show element only on desktop
    @media (min-width: $bp-desktop) {
      display: block;
    }
  }
}
