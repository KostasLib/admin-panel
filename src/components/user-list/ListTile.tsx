import React from 'react';
import cn from 'classnames';
import { IUser } from '../../interfaces/IUser';
import { ListTileText } from './ListTileText';
import s from './ListTile.module.scss';

interface IProps extends React.HTMLProps<HTMLLIElement> {
  user: IUser;
  selected: boolean;
}
export const ListTile: React.FC<IProps> = props => {
  const {
    user: { name, email, photo },
    selected,
    ...rest
  } = props;

  // classnames
  const listClass = cn(s['list-tile'], { [s['list-tile--selected']]: selected });

  return (
    <li className={listClass} {...rest}>
      <div className={s['avatar']}>
        <img className={s['avatar__image']} src={photo} alt={`${name}'s avatar`} />
      </div>
      <ListTileText primary={name} secondary={email} selected={selected} />
    </li>
  );
};
