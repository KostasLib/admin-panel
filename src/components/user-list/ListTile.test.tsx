import React from 'react';
import { shallow } from 'enzyme';
import { ListTile } from './ListTile';

describe('The <ListTile /> component', () => {
  const user = {
    id: '5c093af1c6ee9117a581c7d6',
    photo: 'https://randomuser.me/api/portraits/men/40.jpg',
    name: 'Bates Washington',
    company: 'ZOLAREX',
    email: 'bates.washington@zolarex.io',
    phone: '+1 (915) 447-2207',
    address: '958 Brevoort Place, Ona, Maine, 2433',
  };

  it("renders a <li /> element that contains a user's avatar and details", () => {
    const component = shallow(<ListTile user={user} />);
    expect(component).toMatchSnapshot();
  });

  it('passes the selected prop to its ListTileText child', () => {
    const component = shallow(<ListTile user={user} selected />);
    expect(component).toMatchSnapshot();
  });
});
