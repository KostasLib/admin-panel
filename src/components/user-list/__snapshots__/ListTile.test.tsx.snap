// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <ListTile /> component passes the selected prop to its ListTileText child, if true 1`] = `
<li
  className="list-tile list-tile--selected"
>
  <div
    className="avatar"
  >
    <img
      alt="Bates Washington's avatar"
      className="avatar__image"
      src="https://randomuser.me/api/portraits/men/40.jpg"
    />
  </div>
  <ListTileText
    primary="Bates Washington"
    secondary="bates.washington@zolarex.io"
    selected={true}
  />
</li>
`;

exports[`The <ListTile /> component renders a <li /> element that contains a user's avatar and details 1`] = `
<li
  className="list-tile"
>
  <div
    className="avatar"
  >
    <img
      alt="Bates Washington's avatar"
      className="avatar__image"
      src="https://randomuser.me/api/portraits/men/40.jpg"
    />
  </div>
  <ListTileText
    primary="Bates Washington"
    secondary="bates.washington@zolarex.io"
  />
</li>
`;
